core/src/main/scala-3/sttp/tapir/macros/SchemaMacros.scala

```diff
@@ -252,9 +252,13 @@ private[tapir] object SchemaCompanionMacros {
 
       val subclassesSchemas: List[Expr[(String, Schema[_])]] = subclasses.map(subclass =>
         TypeIdent(subclass).tpe.asType match {
-          case '[f] =>
-            val subSchema = Expr.summon[Schema[f]].get
-            '{ ${ Expr(subclass.name) } -> Schema.wrapWithSingleFieldProduct($subSchema)($conf) }
+          case t @ '[f] =>
+            val opSubSchema = Expr.summon[Schema[f]]
+            if (opSubSchema.isEmpty) {
```

Comment:
nicer:

```
Expr.summon[Schema[f]] match
  case None => report.errorAndAbort(...)
  case Some(...) => ...
```

Reply:
(ot only nicer but in general less error prone, as you can't get it wrong later - generally using `.get` is discouraged

Reply:
Great suggestion. Applied. Thanks!